File: 3185-count-pairs-that-form-a-complete-day-ii/3185-count-pairs-that-form-a-complete-day-ii.cpp
class Solution {
public:
    long long countCompleteDayPairs(vector<int>& hours) {
          long long ans = 0;
    unordered_map<int, int> ump;
    for(auto h: hours){
        ans += ump[(24 - h%24)%24];
        ump[h%24]++;
    }
    return ans;
    }
};

File: 3185-count-pairs-that-form-a-complete-day-ii/3185-count-pairs-that-form-a-complete-day-ii.java
class Solution {
    public long countCompleteDayPairs(int[] hours) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        long ans = 0;
        for (int i=0;i<hours.length;i++) {
            int num = hours[i];
            int mod = num % 24;
            int rem = (24 - mod) % 24;
            ans += mp.getOrDefault(rem, 0);
            mp.put(mod, mp.getOrDefault(mod, 0) + 1);
        }
        return ans;
    }
}